src/Raven.Pal/src/win/pager.c

```diff
@@ -119,13 +117,17 @@ int32_t rvn_lock_memory(int32_t open_flags, void *mem, int64_t size, int32_t *de
     for (int i = 0; i < 10; ++i)
     {
         if (VirtualLock(mem, (SIZE_T)size) ||
-            open_flags & OPEN_FILE_DO_NOT_CONSIDER_MEMORY_LOCK_FAILURE_AS_CATASTROPHIC_ERROR)
+            handle->global_state->open_flags & OPEN_FILE_DO_NOT_CONSIDER_MEMORY_LOCK_FAILURE_AS_CATASTROPHIC_ERROR)
         {
+            // note that we *explicitly* account for locked memory even if we failed to do so
+            // if we don't consider this as catastrophic error, because otherwise accounting 
+            // for its removal is really complicated
+            g_locked_memory_callback(size, handle->global_state->file_path);
             rc = SUCCESS;
```

Comment:
Can you explain a bit more what this mean?

Reply:
Basically, you may run in a mode that doesn't allow to lock memory
In that case, even though the OS didn't lock the memory, our own accounting would.
This is because it would be much more complex for us to keep track of "locked memory in theory, but not really" and not for accounting purposes.

Note that this usually failing to lock memory is a fatal error for us, and you need to configure this especially.

Reply:
It's there, that is what the comment is saying :-)

Reply:
I mean the context of this... Now I understand what that comment means :D